juju/juju
# feat: prevent agent bouncing

4.0 removed the ability to configure a fan network. This is guarded by a model migration pre-check. Attempting to migrate a model with it set, will fail with a user specific error to help them fix the problem.

To prevent the fanconfigurer worker from bouncing when it is migrated, we need to handle the error when that happens.





## Checklist




## QA steps

See [PR](https://github.com/juju/juju/pull/17718) for existing steps to ensure that a fan-config can not be migrated.

Juju bootstrap the following PR https://github.com/juju/juju/pull/17890

```sh
$ juju bootstrap lxd dst40
```

Then bootstrap the change set:

```sh
$ juju bootstrap lxd src36
$ juju add-model moveme
$ juju deploy juju-qa-test
$ juju migrate src36:moveme dst40
```

Switch to dst40 and ensure that there are no errors concerning the fanconfigurer.

```sh
$ juju debug-log dst40:moveme
```

## Links

**Jira card:** [JUJU-6472](https://warthogs.atlassian.net/browse/JUJU-6472)



[JUJU-6472]: https://warthogs.atlassian.net/browse/JUJU-6472?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
